src/streamlink/plugins/nrk.py

```diff
@@ -1,64 +1,82 @@
+        'direkte': 'channel',
+        'serie': 'program',
+        'program': 'program',
+        'podkast': 'podcast',
+    }
 
-_mediaelement_schema = validate.Schema({
-    "mediaUrl": validate.url(
-        scheme="http",
-        path=validate.endswith(".m3u8")
-    )
-})
+    _url_re = re.compile(r"https?://(?:tv|radio)\.nrk\.no/(program|direkte|serie|podkast)(?:/.+)?/([^/]+)")
 
+    _playable_schema = validate.Schema(validate.any(
+        {
+            "playable": {
+                "assets": [{
+                    "url": validate.url(),
+                    "format": str,
+                }],
+            },
+            "statistics": {
+                validate.optional("luna"): validate.any(None, {
+                    "data": {
+                        "title": str,
+                    },
+                }),
+            },
+        },
+        {
+            "nonPlayable": {
+                "reason": str,
+            },
+        },
+    ))
 
-class NRK(Plugin):
     @classmethod
-    def can_handle_url(self, url):
-        return _url_re.match(url)
+    def can_handle_url(cls, url):
+        return cls._url_re.match(url) is not None
 
     def _get_streams(self):
-        # Get the stream type from the url (tv/radio).
-        stream_type = _url_re.match(self.url).group(1).upper()
-        cookie = {
-            "NRK_PLAYER_SETTINGS_{0}".format(stream_type): COOKIE_PARAMS
-        }
-
-        # Construct API URL for this program.
-        baseurl = self.session.http.get(self.url, cookies=cookie, schema=_schema)
-        program_id = _id_re.search(self.url).group(1)
+        # Construct manifest URL for this program.
+        program_type, program_id = self._url_re.match(self.url).groups()
+        manifest_type = self._program_type_map.get(program_type)
+        if manifest_type is None:
+            log.error(f"Unknown program type '{program_type}'")
+            return None
+        manifest_url = urljoin(self._psapi_url, f"playback/manifest/{manifest_type}/{program_id}")
 
         # Extract media URL.
-        json_url = urljoin(baseurl, "mediaelement/{0}".format(program_id))
-        res = self.session.http.get(json_url, cookies=cookie)
-        media_element = self.session.http.json(res, schema=_mediaelement_schema)
-        media_url = media_element["mediaUrl"]
+        res = self.session.http.get(manifest_url)
+        manifest = self.session.http.json(res, schema=self._playable_schema)
+        if 'nonPlayable' in manifest:
+            reason = manifest["nonPlayable"]["reason"]
+            log.error(f"Not playable ({reason})")
+            return None
+        asset = manifest['playable']['assets'][0]
+
+        # Some streams such as podcasts are not HLS but plain files.
+        if asset['format'] == 'HLS':
+            return HLSStream.parse_variant_playlist(self.session, asset['url'])
+        else:
+            return [(self._get_title(manifest), HTTPStream(self.session, asset['url']))]
```

Comment:
What's the title data here, by the way? `_get_streams` is supposed to return a list of tuples or yield tuples in the `name, stream-instance` format. This name however is not supposed to be an actual title, just the name of the stream quality/variant, like `"1080p60"` for example. If it's an actual title, aka stream metadata, then this title has to be returned from the `get_title` method (`get_category` and `get_author` are also available). Since you're returning an HTTPStream here, the common method is returning `"live"` as stream name.

Reply:
Thanks for pointing this out, I wasn't aware of this. Title and category should now be properly set.